library/dokku_letsencrypt.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# from ansible.module_utils.basic import *
+from ansible.module_utils.basic import AnsibleModule
+import subprocess
+
+DOCUMENTATION = """
+---
+module: dokku_letsencrypt
+short_description: Enable or disable the letsencrypt plugin for a dokku app
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  state:
+    description:
+      - The state of the letsencrypt plugin
+    required: False
+    default: present
+    choices: [ "present", "absent" ]
+    aliases: []
+author: Gavin Ballard
+requirements:
+  - the `dokku-letsencrypt` plugin
+"""
+
+EXAMPLES = """
+- name: Enable the letsencrypt plugin
+  dokku_letsencrypt:
+    app: hello-world
+
+- name: Disable the letsencrypt plugin
+  dokku_letsencrypt:
+    app: hello-world
+    state: absent
+"""
+
+
+def force_list(var):
+    if isinstance(var, list):
+        return var
+    return list(var)
+
+
+def subprocess_check_output(command, split="\n"):
+    error = None
+    output = []
+    try:
+        output = subprocess.check_output(command, shell=True)
+        if isinstance(output, bytes):
+            output = output.decode("utf-8")
+        output = str(output).rstrip("\n")
+        if split is None:
+            return output, error
+
+        output = output.split(split)
+        output = force_list(filter(None, output))
+        output = [o.strip() for o in output]
+    except subprocess.CalledProcessError as e:
+        error = str(e)
+    return output, error
+
+
+def dokku_letsencrypt(data):
+    command = "dokku --quiet letsencrypt:list | awk '{{print $1}}'"
+    response, error = subprocess_check_output(command.format(data["app"]))
+
+    if error:
+        return "0", error
+
+    if data["app"] in response:
+        return "1", error
+
+    return "0", error
+
+
+def dokku_letsencrypt_present(data):
+    is_error = True
+    has_changed = False
+    meta = {"present": False}
+
+    enabled, error = dokku_letsencrypt(data)
+    if enabled == "1":
+        is_error = False
+        meta["present"] = True
+        return (is_error, has_changed, meta)
+
+    command = "dokku --quiet letsencrypt:enable {0}".format(data["app"])
+    try:
+        subprocess.check_call(command, shell=True)
+        is_error = False
+        has_changed = True
+        meta["present"] = True
+    except subprocess.CalledProcessError as e:
+        meta["error"] = str(e)
+
+    return (is_error, has_changed, meta)
+
+
+def dokku_letsencrypt_absent(data=None):
+    is_error = True
+    has_changed = False
+    meta = {"present": True}
+
+    enabled, error = dokku_letsencrypt(data)
+    if enabled == "0":
+        is_error = False
```

Comment:
is that correct?
your `dokku_letsencrypt` function also returns "0" when it encounters an error.

perhaps add a comment here on why it is ok to ignore that.